compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/expression/FirGetClassCallChecker.kt

```diff
@@ -71,8 +77,14 @@ object FirGetClassCallChecker : FirBasicExpressionChecker() {
         get() {
             return this is FirResolvedQualifier ||
                     this is FirResolvedReifiedParameterReference ||
-                    ((this is FirQualifiedAccessExpression) &&
-                            (this.calleeReference as? FirResolvedNamedReference)?.resolvedSymbol is FirTypeParameterSymbol)
+                    safeAsTypeParameterSymbol != null
+        }
+
+    private val FirExpression.safeAsTypeParameterSymbol: FirTypeParameterSymbol?
```

Comment:
Nit: Just my opinion but the name is a little confusing, I think just `asTypeParameterSymbol` is ok.

Reply:
I think `safe` emphasizes the nullable return type. E.g.,

* https://github.com/JetBrains/kotlin/blob/master/compiler/fir/tree/src/org/jetbrains/kotlin/fir/expressions/FirOperation.kt#L36
* https://github.com/JetBrains/kotlin/blob/master/core/util.runtime/src/org/jetbrains/kotlin/utils/addToStdlib.kt#L73
* https://github.com/JetBrains/kotlin/blob/master/compiler/fir/tree/src/org/jetbrains/kotlin/fir/types/FirTypeUtils.kt#L20